libfwupdplugin/fu-hwids-dmi.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2021 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#define G_LOG_DOMAIN "FuContext"
+
+#include "config.h"
+
+#include "fu-context-private.h"
+#include "fu-hwids-private.h"
+#include "fu-string.h"
+
+gboolean
+fu_hwids_dmi_setup(FuContext *ctx, FuHwids *self, GError **error)
+{
+	const gchar *path = g_getenv("FWUPD_SYSFSDMIDIR");
+	g_autofree gchar *path_dmi_class = NULL;
+	struct {
+		const gchar *hwid;
+		const gchar *key;
+	} map[] = {{FU_HWIDS_KEY_BASEBOARD_MANUFACTURER, "board_vendor"},
+		   {FU_HWIDS_KEY_BASEBOARD_PRODUCT, "board_name"},
+		   {FU_HWIDS_KEY_BIOS_VENDOR, "bios_vendor"},
+		   {FU_HWIDS_KEY_BIOS_VERSION, "bios_version"},
+		   {FU_HWIDS_KEY_FAMILY, "product_family"},
+		   {FU_HWIDS_KEY_MANUFACTURER, "sys_vendor"},
+		   {FU_HWIDS_KEY_PRODUCT_NAME, "product_name"},
+		   {FU_HWIDS_KEY_PRODUCT_SKU, "product_sku"},
+		   {FU_HWIDS_KEY_ENCLOSURE_KIND, "chassis_type"},
+		   {NULL, NULL}};
+
+	/* the values the kernel parsed; these are world-readable */
+	if (path != NULL) {
+		path_dmi_class = g_build_filename(path, "dmi", "class", NULL);
```

Comment:
as part of the refactor can we make the subdirectory `"id"` so then this can be a single path check that looks more like a real system?

Reply:
It actually means we can do the opposite and make it two levels flatter in the testdir.